### PR TITLE
[CI] Add workflow to push Docker image to Dockerhub and ECR

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR adds a GitHub Actions workflow to automatically push a new version of this repo's Docker image to both Dockerhub and ECR.
The former is used when running the tooling on your local machine using the [development environment](https://github.com/exercism/development-environment/), the latter is currently used when running the tooling on our staging environment (https://exercism.lol/) but will eventually be used for our production website.

We've sent PRs to tooling repos before related to this functionality, but those PRs added two separate workflows for pushing to Dockerhub and ECR. With this PR, these workflows have been merged into a single workflow, which is much more efficient. If present, the old workflows will be removed in this PR.

See https://github.com/exercism/v3/issues/2969

### CODE REVIEW

**Review by ynfle — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
@ynfle It looks like you don't have review permissions for this repo. Would you be interested in having those permissions?

File: .github/workflows/docker.yml
Comment (ee7) [inline .github/workflows/docker.yml:8]:
```suggestion
  multiple-registries:
```
Non-blocking comment: I think this name could be better.

File: .github/workflows/docker.yml
Comment (ee7) [inline .github/workflows/docker.yml:35]:
```suggestion
      - name: Log in to ECR
```
Same nitpick.

**Review by ee7 — APPROVED** [review]
Nice - this PR also upgrades `docker/build-push-action` to v2.

And it's great that this uses `actions/cache` rather than `satackey/action-docker-layer-caching`, which caused some CI errors previously (see https://github.com/exercism/nim-test-runner/runs/1382436417).

Approved, conditional on the `docker/build-push-action` version number being fixed.

**Comment by ynfle** [comment]
> @ynfle It looks like you don't have review permissions for this repo. Would you be interested in having those permissions?

Which permissions? Didn't I used to have permissions?

File: .github/workflows/docker.yml
Comment (ErikSchierboom) [reply .github/workflows/docker.yml:5]: We'd prefer to have both branches here, as it will prevent us from breaking things when we start using main. We can indeed rename later.

**Comment by ErikSchierboom** [comment]
Hmmm, there is something weird here. Both of you have approved, but they don't count as approving reviews. @iHiD could you check if the permissions are correct for @ee7 and @ynfle?

**Comment by ee7** [comment]
> Hmmm, there is something weird here. Both of you have approved, but they don't count as approving reviews.

Isn't it just because the CODEOWNER file contains this?
https://github.com/exercism/nim-test-runner/blob/aba25f9aef205ddb08fe94cd62fb602af283b160/.github/CODEOWNERS#L1

**Review by iHiD — APPROVED** [review]

**Pull request merged**
